inference-engine/src/inference_engine/include/openvino/runtime/gpu/ocl.hpp

```diff
@@ -0,0 +1,237 @@
+     * @return `cl::Image2D` object
+     */
+    operator cl::Image2D() {
+        return cl::Image2D(get(), true);
+    }
+};
+
+/**
+ * @brief This class represents an abstraction for GPU plugin remote context
+ * which is shared with OpenCL context object.
+ * The plugin object derived from this class can be obtained either with
+ * GetContext() method of Executable network or using CreateContext() Core call.
+ */
+class ClContext : public RemoteContext {
+    using RemoteContext::create_tensor;
+    using RemoteContext::RemoteContext;
+
+public:
+    /**
+     * @brief Checks that type defined runtime paramters are presented in remote object
+     * @param remote_context remote context to check
+     */
+    static void type_check(const RemoteContext& remote_context) {
+        remote_type_check(remote_context.get_params(),
+                          {{GPU_PARAM_KEY(OCL_CONTEXT), {}},
+                           {GPU_PARAM_KEY(CONTEXT_TYPE), {GPU_PARAM_VALUE(OCL), GPU_PARAM_VALUE(VA_SHARED)}}});
+    }
+
+    /**
+     * @brief Constructs context object from user-supplied OpenCL context handle
+     * @param core A reference to Inference Engine Core object
+     * @param deviceName A name of device to create a remote context for
+     * @param ctx A OpenCL context to be used to create shared remote context
+     */
+    ClContext(Core& core, std::string deviceName, cl_context ctx) {
+        ParamMap contextParams = {{GPU_PARAM_KEY(CONTEXT_TYPE), GPU_PARAM_VALUE(OCL)},
+                                  {GPU_PARAM_KEY(OCL_CONTEXT), static_cast<gpu_handle_param>(ctx)}};
+        *this = core.create_context(deviceName, contextParams).as<ClContext>();
+    }
+
+    /**
+     * @brief Returns the underlying OpenCL context handle.
+     * @return `cl_context`
+     */
+    cl_context get() {
+        return static_cast<cl_context>(get_params().at(GPU_PARAM_KEY(OCL_CONTEXT)).as<gpu_handle_param>());
+    }
+
+    /**
+     * @brief OpenCL context handle conversion operator for the ClContext object.
+     * @return `cl_context`
+     */
+    operator cl_context() {
+        return get();
+    }
+
+    /**
+     * @brief Standard Khronos cl::Context wrapper conversion operator for the ClContext object.
+     * @return `cl::Context` object
+     */
+    operator cl::Context() {
+        return cl::Context(get(), true);
+    }
+
+    /**
+     * @brief This function is used to construct a NV12 compound blob object from two cl::Image2D wrapper objects.
+     * The resulting compound contains two remote blobs for Y and UV planes of the surface.
+     * @param nv12_image_plane_y cl::Image2D object containing Y plane data.
+     * @param nv12_image_plane_uv cl::Image2D object containing UV plane data.
+     * @return A RemoteTensor instance
+     */
+    RemoteTensor create_tensor_nv12(const cl::Image2D& nv12_image_plane_y, const cl::Image2D& nv12_image_plane_uv) {
```

Comment:
can we removed since we don't have NV12 blobs in new API

Reply:
I suggest to return pair of tensors